### PR TITLE
add stdcm openapi spec (core + api)

### LINKED ISSUE #1413: Design STDCM module to current production simulation service
## Associated Feature / User Story
F3: STDCM

## Description
Design application programming interface for STDCM consumers 

## Goal

Define how OSRD interact with STDCM algorithm 

## Acceptance criteria

- [ ] Having a reviewed and validated pull request for Open API changes

## Story points

- [ ] 1
- [x] 2
- [ ] 3
- [ ] 5
- [ ] 8

## Definition of ready

- [x] The enabler story is well defined
- [x] Acceptance criterias are defined
- [ ] The individual responsible for validating the enbaler story is assigned
- [x] Does not depend on an missing feature
- [x] Negociable with the implementation team
- [x] Valuable to the end users
- [x] Estimable by the implementation team
- [x] As small as it can be
- [x] Testable by the implementation team

## Definition of done

- [ ] Acceptance criteria are met
- [ ] Automated tests are written against acceptance criteria
- [ ] Story ok-ed by Product Owner
- [ ] Documentation updated

### PR DESCRIPTION
**Pull request by eckter**
Wait for implementation before merging

### CODE REVIEW

**Review by flomonster — DISMISSED** [review]

File: core/openapi.yaml [hunk 1]
```diff
@@ -457,24 +486,59 @@ components:
...
+              start_points:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Waypoint"
+                  minimum: 1
+              end_points:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Waypoint"
+                  minimum: 1
```
Comment (multun) [inline core/openapi.yaml]: I'm not sure I get it, what's that?

Comment (eckter) [reply core/openapi.yaml]: Start and end points for the pathfinding, it's the same as for the normal pathfinding but without intermediate points

File: api/openapi.yaml [hunk 2]
```diff
@@ -901,17 +933,47 @@ components:
...
+            type: integer
+        timetable:
+          type: integer
+          description: timetable ID
+        start_time:
+          type: number
+          format: double
+        end_time:
+          type: number
+          format: double
```
Comment (flomonster) [inline api/openapi.yaml:963]: Why do we have both a start and end time?

File: core/openapi.yaml [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
     name: GNU GPLv3
     url: https://www.gnu.org/licenses/gpl-3.0.fr.html
   version: 0.1.0
 servers:
-- url: http://gateway.dgexsol.fr/osrd/
-  description: Production server
-- url: http://gateway.staging.dgexsol.fr/osrd/
-  description: Staging server
+  - url: http://gateway.dgexsol.fr/osrd/
+    description: Production server
```
Comment (flomonster) [inline core/openapi.yaml]: We don't have production server yet. Bu we have the dev one:

```suggestion
  - url: http://gateway.dev.dgexsol.fr/osrd/
    description: Dev server
```

File: core/openapi.yaml [hunk 4]
```diff
@@ -457,24 +490,62 @@ components:
...
+                format: double
+              end_occupancy_time:
+                type: number
+                format: double
+        start_time:
+          type: number
+          format: double
+        end_time:
+          type: number
+          format: double
```
Comment (flomonster) [inline core/openapi.yaml:530]: Same question here..

**Review by flomonster — CHANGES_REQUESTED** [review]

File: api/openapi.yaml (hunk 2 above)
Comment (eckter) [reply api/openapi.yaml:963]: Eventually we'll be able to specify either one as input

It should be "either one of those", I'll edit the specs

File: api/openapi.yaml [hunk 5]
```diff
@@ -698,6 +700,36 @@ paths:
...
+      responses:
+        200:
+          description: Simulation result
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - properties:
+                      simulation:
+                        $ref: "#/components/schemas/TrainScheduleResult"
```
Comment (eckter) [inline api/openapi.yaml:722]:
```suggestion
                        $ref: "#/components/schemas/TrainScheduleResultData"
```

File: api/openapi.yaml [hunk 6]
```diff
@@ -698,6 +700,36 @@ paths:
                   properties: { ... }
                   required: [...]
                   definitions: { ... }
+  /stdcm:
```
Comment (eckter) [inline api/openapi.yaml]:
```suggestion
  /stdcm/:
```

**Review by flomonster — APPROVED** [review]

**Pull request merged**